recipe-client-addon/lib/NormandyDriver.jsm

```diff
@@ -156,6 +156,68 @@ this.NormandyDriver = function(sandboxManager) {
       sandboxManager.removeHold(`setTimeout-${token}`);
     },
 
+    /*
+     * Return a promise that resolves to an Addon ID if installation is successful.
+     */
+    installAddon(url) {
+        /* Install an addon.  Note that the AddonManager can only handle
+         * file:// and downloadable URLs.
+         */
+
+      var installObjectCallback = (installObject, resolve, reject) => {
+        installObject.install();
+        installObject.addListener({
+          onInstallEnded: (addonInstall, addon) => {
+            let addonId = addon.id;
+            resolve(addonId);
+          },
+          onInstallFailed: (addonInstall) => {
+            reject(`AddonInstall error code: [${addonInstall.error}]`);
+          },
+        });
+      };
+
+      function installObjectPromise(installObject) {
```

Comment:
All this does is wrap `installObjectCallback`, which seems to have no other callers. Just put the contents of `installObjectCallback` straight into the promise function in here, then you don't need to pass reject/resolve around, either.